### PR TITLE
RIA-8211 update tribunal decision rule 31 AIP timeline

### PR DESCRIPTION
**Pull request by DennisHoV1**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RIA-8211

### Change description ###
- update the content of "do this next" section when trigger "updateTribunalDecision" event
- update the content of "Completed" secton  when trigger "updateTribunalDecision" event
- updated the unit test for all changes

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: app/utils/event-deadline-date-finder.ts [hunk 1]
```diff
@@ -49,12 +53,19 @@ function getFormattedEventHistoryDate(history: HistoryEvent[], eventTagToLookFor
  * @param req the request containing the appeal information
  * @returns deadline for the appeallnt to respond to Judge's decision
  */
-function getDueDateForAppellantToRespondToJudgeDecision(req: Request) {
-  const finalDecisionAndReasonsPdfDoc = req.session.appeal.finalDecisionAndReasonsDocuments.find(doc => doc.tag === 'finalDecisionAndReasonsPdf');
+function getDueDateForAppellantToRespondToJudgeDecision(req: Request, ftpaSetAsideFeatureEnabled: Boolean = false) {
```
Comment (MiguelMartinRojas) [inline app/utils/event-deadline-date-finder.ts:56]: I am missing tests for this file

**Review by MiguelMartinRojas — CHANGES_REQUESTED** [review]

**Review by MiguelMartinRojas — APPROVED** [review]

File: app/utils/timeline-utils.ts [hunk 2]
```diff
@@ -144,6 +145,54 @@ function getDirectionHistory(req: Request): any[] {
...
+    return [];
+  }
+}
+
+function getUpdateTribunalDecisionDocumentHistory(req: Request, ftpaSetAsideFeatureEnabled: boolean): any[] {
+  if (isUpdateTribunalDecideWithRule31(req, ftpaSetAsideFeatureEnabled) && req.session.appeal.decisionAndReasonDocsUpload) {
+
+    const coverLetterDocument = req.session.appeal.finalDecisionAndReasonsDocuments.find(doc => doc.tag === 'updatedDecisionAndReasonsCoverLetter');
+
+    return [{
```
Comment (ThomasKKC) [inline app/utils/timeline-utils.ts:182]: As discussed, lets review this together. See if we can use the UPDATE_TRIBUNAL_DECISION event.

File: types/index.d.ts [hunk 3]
```diff
@@ -159,6 +159,10 @@ interface Appeal {
   addendumEvidence?: AdditionalEvidenceDocument[];
   pcqId?: string;
   isDecisionAllowed?: string;
+  updateTribunalDecisionList?: string;
+  typesOfUpdateTribunalDecision?: DynamicList;
```
Comment (ThomasKKC) [inline types/index.d.ts:163]: Do we need this field? think we can omit it.

File: types/caseData.d.ts [hunk 4]
```diff
@@ -136,6 +136,10 @@ interface CaseData {
   witnessDetails?: Collection<WitnessDetails>[];
   interpreterLanguage?: Collection<InterpreterLanguage>[];
   isDecisionAllowed?: string;
+  updateTribunalDecisionList?: string;
+  typesOfUpdateTribunalDecision?: DynamicList;
```
Comment (ThomasKKC) [inline types/caseData.d.ts:140]: Do we need this field? think we can omit it.

File: app/utils/timeline-utils.ts [hunk 5]
```diff
@@ -144,6 +145,54 @@ function getDirectionHistory(req: Request): any[] {
...
+    }];
+  } else {
+    return [];
+  }
+}
+
+function getUpdateTribunalDecisionDocumentHistory(req: Request, ftpaSetAsideFeatureEnabled: boolean): any[] {
+  if (isUpdateTribunalDecideWithRule31(req, ftpaSetAsideFeatureEnabled) && req.session.appeal.decisionAndReasonDocsUpload) {
+
+    const coverLetterDocument = req.session.appeal.finalDecisionAndReasonsDocuments.find(doc => doc.tag === 'updatedDecisionAndReasonsCoverLetter');
```
Comment (ThomasKKC) [inline app/utils/timeline-utils.ts:180]: Let's use history event for the date

**Review by ThomasKKC — CHANGES_REQUESTED** [review]

File: types/index.d.ts (hunk 3 above)
Comment (DennisHoV1) [reply types/index.d.ts]: removed the case field "typesOfUpdateTribunalDecision" and "decisionAndReasonDocsUpload"

File: types/caseData.d.ts (hunk 4 above)
Comment (DennisHoV1) [reply types/caseData.d.ts]: removed the case field "typesOfUpdateTribunalDecision" and "decisionAndReasonDocsUpload"

File: app/utils/timeline-utils.ts (hunk 5 above)
Comment (DennisHoV1) [reply app/utils/timeline-utils.ts]: changed to use the datetime of updateTribunalDecision event from the history

**Pull request merged**